### ISSUE TITLE
Broker username and password

### ISSUE THREAD

**Issue by navi7** [reecetech/pactman#52]
Our Pact broker is behind username and password, so we would need the missing pieces that are mentioned here: https://github.com/reecetech/pactman/blob/master/pactman/verifier/command_line.py#L12

We would be interested in doing the work and providing a PR if that would be something you would be OK with?

**Comment by richard-reece** [comment]
Yep, go for it, thanks!

**Comment by amir-hadi** [comment]
@navi7 I believe another workaround could be (until a PR is merged) is to use HTTP auth in the URL, like `http://pact:broker@pact-broker.com/`

**Comment by navi7** [comment]
@amir-hadi I have implemented it (and we are already using a build from our fork in our CI), but I have not written any tests :shame: so I didn't create the PR.

**Comment by amir-hadi** [comment]
@navi7 Also my workaround is not working, as it doesn't propagate to other requests. I will have a look on your fork.

**Comment by navi7** [comment]
It's nothing smart, most of the code is to get the credentials from the command line or env vars. Then I just pass them to the Navigator thingie (which is the first time I see/use the `restnavigator`).

**Comment by richard-reece** [comment]
Yeah, the restnavigator stuff is a bit funky, but it beats the heck outta trying to navigate HAL manually, believe me :-(

**Comment by mcmont** [comment]
Thanks @navi7 for adding the broker credentials arguments! If there's any chance that this could be merged into the main repo soon I'd be very grateful. Thanks.

**Comment by richard-reece** [comment]
I don't have a way to sanity check the implementation against a protected broker so I'm just waiting on a PR.

**Comment by richard-reece** [comment]
@navi7, @amir-hadi  and @mcmont I'd appreciate if you had a look at PR #64 which implements basic auth!

**Comment by navi7** [comment]
@richard-reece Thanks for this!
